internal/controllers/gateway/httproute_controller_envtest_test.go

```diff
@@ -0,0 +1,354 @@
+					CacheNamespace: ns.Name,
+				},
+			})
+
+			gwc := gatewayv1beta1.GatewayClass{
+				Spec: gatewayv1beta1.GatewayClassSpec{
+					ControllerName: gateway.GetControllerName(),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: ns.Name,
+					Name:      uuid.NewString(),
+					Annotations: map[string]string{
+						"konghq.com/gatewayclass-unmanaged": "placeholder",
+					},
+				},
+			}
+			require.NoError(t, client.Create(ctx, &gwc))
+
+			gw := gatewayv1beta1.Gateway{
+				Spec: gatewayv1beta1.GatewaySpec{
+					GatewayClassName: gatewayv1beta1.ObjectName(gwc.Name),
+					Listeners: []gatewayv1beta1.Listener{
+						{
+							Name:     gatewayv1beta1.SectionName("http"),
+							Port:     gatewayv1beta1.PortNumber(80),
+							Protocol: gatewayv1beta1.HTTPProtocolType,
+							AllowedRoutes: &gatewayv1beta1.AllowedRoutes{
+								Namespaces: &gatewayv1beta1.RouteNamespaces{
+									From: lo.ToPtr(gatewayv1beta1.NamespacesFromAll),
+								},
+							},
+						},
+					},
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: ns.Name,
+					Name:      uuid.NewString(),
+				},
+			}
+			require.NoError(t, client.Create(ctx, &gw))
+
+			gwOld := gw.DeepCopy()
+			gw.Status = gatewayv1beta1.GatewayStatus{
+				Addresses: []gatewayv1beta1.GatewayAddress{
+					{
+						Type:  lo.ToPtr(gatewayv1beta1.IPAddressType),
+						Value: "10.0.0.1",
+					},
+				},
+				Conditions: []metav1.Condition{
+					{
+						Type:               "Ready",
+						Status:             metav1.ConditionTrue,
+						Reason:             "Ready",
+						LastTransitionTime: metav1.Now(),
+						ObservedGeneration: gw.Generation,
+					},
+					{
+						Type:               "Accepted",
+						Status:             metav1.ConditionTrue,
+						Reason:             "Accepted",
+						LastTransitionTime: metav1.Now(),
+						ObservedGeneration: gw.Generation,
+					},
+					{
+						Type:               "Programmed",
+						Status:             metav1.ConditionTrue,
+						Reason:             "Programmed",
+						LastTransitionTime: metav1.Now(),
+						ObservedGeneration: gw.Generation,
+					},
+				},
+				Listeners: []gatewayv1beta1.ListenerStatus{
+					{
+						Name: gatewayv1beta1.SectionName("http"),
+						Conditions: []metav1.Condition{
+							{
+								Type:               "Accepted",
+								Status:             metav1.ConditionTrue,
+								Reason:             "Accepted",
+								LastTransitionTime: metav1.Now(),
+							},
+							{
+								Type:               "Ready",
+								Status:             metav1.ConditionTrue,
+								Reason:             "Ready",
+								LastTransitionTime: metav1.Now(),
+							},
+						},
+						SupportedKinds: []gatewayv1beta1.RouteGroupKind{
+							{
+								Group: lo.ToPtr(gatewayv1beta1.Group(gatewayv1beta1.GroupVersion.Group)),
+								Kind:  "HTTPRoute",
+							},
+						},
+					},
+				},
+			}
+			require.NoError(t, client.Status().Patch(ctx, &gw, ctrlclient.MergeFrom(gwOld)))
```

Comment:
Since we are using envtest, can we run the gateway controller along with the HTTPRoute one to have this update done on our behalf by it?

Reply:
I'm on the fence too honestly. Improving isolation is something valuable, while on the other hand, I think that running the `Gateway` Controller would make the test way clearer and neater. I'll leave the decision to you, I don't want to push to a specific direction, as each one has pros and cons.

Reply:
👍 Let's leave it at that for now. I'm happy to discuss it again but the isolation is the biggest selling point for me.